src/car.rs

```diff
@@ -0,0 +1,83 @@
+use crate::error::Error;
+
+pub struct Car {
+    pub name: String,
+    pub distance: u32,
+}
+
+impl Car {
+    pub fn new(name: &str) -> Result<Car, Error> {
+        let name = name.trim();
+        let name_char_length = name.chars().count();
+        if name_char_length == 0 {
+            return Err(Error::BlankCarName);
+        }
+        if name_char_length > 5 {
+            return Err(Error::CarNameOverFiveCharacters);
+        }
+
+        Ok(Car {
+            name: String::from(name),
+            distance: 0,
+        })
+    }
+
+    pub fn play(&self, number: i8) -> Car {
```

Comment:
Q) Is there any reason to create a new object?
To read the code, I'm curious as to why you create a new object instead of modifying the original.